src/librbd/Journal.cc

```diff
@@ -23,6 +23,17 @@
 
 namespace librbd {
 
+template<typename T>
```

Comment:
Yep, this will become cleaner, `although handle_metadata_updated` is already private, but now I can remove the friend class declaration.